src/crystal/system/win32/process.cr

```diff
@@ -203,13 +205,22 @@ struct Crystal::System::Process
 
     process_info = LibC::PROCESS_INFORMATION.new
 
-    command_args = ::Process.quote_windows(command_args) unless command_args.is_a?(String)
+    if command_args.is_a?(String)
+      result = LibC.CreateProcessW(
+        CMD_EXE, System.to_wstr(%(/c "#{command_args}")), nil, nil, true, LibC::CREATE_UNICODE_ENVIRONMENT,
+        make_env_block(env, clear_env), chdir.try { |str| System.to_wstr(str) },
+        pointerof(startup_info), pointerof(process_info)
+      )
+    else
+      command_args = ::Process.quote_windows(command_args)
+      result = LibC.CreateProcessW(
+        nil, System.to_wstr(command_args), nil, nil, true, LibC::CREATE_UNICODE_ENVIRONMENT,
+        make_env_block(env, clear_env), chdir.try { |str| System.to_wstr(str) },
+        pointerof(startup_info), pointerof(process_info)
+      )
+    end
```

Comment:
Both calls seem to be identical except for the first two parameters. So I figure it might be easier to understand the code with less duplication.
```suggestion
    if command_args.is_a?(String)
      command_args = %(/c "#{command_args}")
      app_name = CMD_EXE
    else
      command_args = ::Process.quote_windows(command_args)
    end

    result = LibC.CreateProcessW(
        app_name, System.to_wstr(command_args), nil, nil, true, LibC::CREATE_UNICODE_ENVIRONMENT,
        make_env_block(env, clear_env), chdir.try { |str| System.to_wstr(str) },
        pointerof(startup_info), pointerof(process_info)
      )
```

Reply:
@HertzDevil WDYT?